### DISCUSSION TITLE
Upload folders on web

### DISCUSSION THREAD

**Discussion by ghost** [immich-app/immich#5938] [Feature Request]
### The feature

When I click the upload button ,the system open a windows that I can only choose the files,but not the folder.Wish Can add a new feature that let us upload folders to make things easy.

When we clike to chosse a folder ,the system can autoly search the files to upload..tks
![image](https://github.com/immich-app/immich/assets/50078180/7ab66b3d-f705-44ee-b151-c3341fd6d1ee)

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by daschillinger** [comment]
atm this is done via the CLI, if this is a workaround for you
[https://immich.app/docs/features/command-line-interface](https://immich.app/docs/features/command-line-interface)

**Comment by mdafer** [comment]
+1 for this feature, it would be much nicer and easier for users with no cli access to do it!

**Reply by sirweazel** [reply]
I was about to open as a feature request, then i found someone already did, so i hit the up arrow.  For now, i use shift select and highlight all the photos in a given directory. I think i do need to read up on the command line interface.

**Comment by ghost** [comment]
Uploading via the command line is not good, if my image library is on a computer, then I have to upload the file to the server first, and then I need to have immich scan it. Then I need to delete the file I just uploaded to the server.

**Comment by smitdylan2001** [comment]
This is really needed. I have a huge backup to be uploaded, but it is all in folders and subfolders. For someone experienced with the code base this does not seem like a huge time investment for a big feature

**Reply by bo0tzz** [reply]
If you have a large amount of stuff to upload you should probably use the CLI instead: https://immich.app/docs/features/command-line-interface/

**Comment by felixconsulting** [comment]
I was able to do this, I dragged a bunch of folders from konqueror to brave browser with immich open, and it uploaded all the images in them. now i can't do it again :(

**Reply by leeli0** [reply]
<img width="1868" height="286" alt="Screenshot From 2026-07-05 22-00-21" src="https://github.com/user-attachments/assets/3d77224c-41c6-43c5-a1cb-6010775a3370" />
I tried, it worked. Dragging the folder into the browser tab which opend immich. But this method can only handle folders have dozens of files. For bulk uploads cli is the right choice.
